### ISSUE TITLE
StackPanel Content not Centered Within Button

### ISSUE THREAD

**Issue by robloo** [unoplatform/uno#7620]
### Current behavior

Certain Button Content is not centered. This is even specified in XAML with `HorizontalContentAlignment` just to be sure but the default is also `Center`. This appears to be related to the StackPanel which is known to have many issues. 

Note: StackPanel needs a port from upstream WinUI ASAP like Grid.

![image](https://user-images.githubusercontent.com/17993847/145271649-5cc5f48c-2ee2-4198-804d-98c3ec524dc2.png)

```xaml
                <Button
                    x:Name="NextButton2"
                    MinWidth="100"
                    HorizontalContentAlignment="Center">
                    <StackPanel
                        VerticalAlignment="Stretch"
                        HorizontalAlignment="Stretch"
                        Orientation="Horizontal">
                        <TextBlock
                            x:Name="NextButton2TextBlock"
                            Text="Next"
                            TextAlignment="Left"
                            VerticalAlignment="Center"
                            Margin="0,0,3,0" />
                    </StackPanel>
                </Button>
                <Button
                    x:Name="NextButton3"
                    MinWidth="100"
                    Content="Next"
                    HorizontalContentAlignment="Center">
                </Button>
```

### Expected behavior

All content should be centered by default.

### How to reproduce it (as minimally and precisely as possible)

Repro App:

[TestApp.zip](https://github.com/unoplatform/uno/files/7678390/TestApp.zip)

### Workaround

Perhaps try a different container/panel such as Grid

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.0.9

### Affected platforms

Android

### IDE

Visual Studio 2022

**Comment by jeromelaban** [comment]
The issue is not with `StackPanel`, but rather how button manages its own children. You can replace the `StackPanel` with a Grid and you'll see the same result.

**Comment by jeromelaban** [comment]
@carldebilly do you know what happens here ?

**Comment by carldebilly** [comment]
Playground: https://playground.platform.uno/#161c3915

Looks related to the way the "arrange" is build in Uno. My intuition would be it should take into consideration the desiredSize and readjust according to the actual result from ArrangeOverride(). This mechanism is pretty different in Uno and could explain the behavior observed in this issue.

To investigate...

**Comment by robloo** [comment]
In the playground, first sample, I can delete the `HorizontalAlignment="Stretch"` from `<StackPanel Orientation="Horizontal" HorizontalAlignment="Stretch">` and make it work. This shows two things:

 1. Measure/arrange is behaving differently on WASM from Android. I would normally expect these layout-related code paths to be shared and generic for all platforms.
 2. StackPanel is not sizing to the child TextBox. This is a bug with StackPanel as I do think it always sizes to children in UWP.

Bottom line: Wasm appears to behave differently from Android.

**Comment by Youssef1313** [comment]
If this is Android/iOS only, then I would suspect it's because of ContentPresenter bypass.

**Comment by Youssef1313** [comment]
https://github.com/unoplatform/uno/pull/18261 may have an impact

**Comment by MartinZikmund** [comment]
We should validate whether this works correctly in Skia, in which case it can be closed @Jen-Uno

**Comment by rajamatt** [comment]
Looks good with Skia. Using latest dev uno.sdk version 6.3.0-dev.84.

---

Tested with:
Templates/SDK: 6.3.0-dev.84 (which uses Uno version 6.3.0-dev.177)
Platform(s): Desktop, Android, Wasm
IDE: VS 2022 - 17.14.12
OS: Windows
